Automattic/sensei
# Add logging for initial publish of a course

Closes #2650 

Adds logging for when a course is initially published but does not log if the course is unpublished and published again later.

## Implementation note

There is a check here for meta that is added by WCPC. I don't really like it being in this codebase since it seems like it should be in WCPC itself. However, since it is just meta, there isn't any strong technical reason why this plugin couldn't simply check for it. This keeps it much simpler but introduces a funny dependency.

Open to thoughts on this.

## Testing instructions

_Note: please test this in both the classic editor **and** the block editor._

- Create a course and save it as a draft.
- Add modules to the course.
- Add lessons to the course.
- Publish the course. Ensure that the logged event has the correct attributes (including `-1` for the `product_id`).
- Unpublish the course and publish it again. Ensure that another event is not logged.
- Follow steps 1-3 and then install WCPC and link a product to the course.
- Publish the course, and ensure the `product_id` attribute on the event is correct.
- Create another draft course, but do not include any modules or lessons.
- Publish the course and ensure that the `module_count` and `lesson_count` attributes are `0`.
- Create a course and add modules, lessons, and a product without saving as a draft (set all of these things while on the initial "Add Course" screen). On publishing, ensure that the event is logged, and the properties are set correctly.

### Additional test for existing installations

- On a site with courses that have already been published, switch to this branch.
- Unpublish and publish an existing course.
- Ensure that no event is logged.
